### PR TITLE
Chore/improve search input sql editor

### PR DESCRIPTION
**Pull request by MildTomato**
# UI components for inner sidepanel query items

## PR reverted to less visual changes.

- mostly this is moving code into reusable components 
- inner sidebar components setup ready to be collapsible
- empty states moved to components
- Search input on SQL editor moved below Templates/Quickstarts
- also, search input does not filter Templates and Quick-starts as they are not user items.
  - thought it was weird it was being treated the same.
- Sort option moved inside (visually) into input.
    
Screenshot 2024-03-05 at 3 00 06 PM

image

# old stuff

~~SQL editor sidebar components and sections~~

~~I’ve spent some time over the weekend tidying items up into ./patterns while also trying to make the SQL editor sidebar a bit more useful for searching, especially when you have more items.~~

~~I’ve also used the same patterns from Table Editor so the search filter and items are all shared components.~~

~~Video showing most of it.~~

https://github.com/supabase/supabase/assets/8291514/17a1df01-ad59-49a6-9e2e-5f73a9ecd0e7

### ~~**Things to consider**~~

- ~~Maybe need to use localstorage to store the collapse state of the inner sidebar sections.~~

### Proposals

- ~~**Everything is now components**~~

[](https://github.com/supabase/supabase/blob/chore/improve-search-input-sql-editor/packages/ui-patterns/InnerSideMenu/index.tsx)

- ~~**The inner sidebar is now split up into permanent sections, project, fav and personal**~~
    - ~~They each have their own empty state.~~
        - ~~favorite section wasn’t visible before, I forgot it was a thing.
        I know we might end up dropping this but for the time being it seemed a good kaizen~~    
        image
    - ~~Each section has it’s own search, and potentially own sorting controls.~~
    - ~~They are all collapsible~~
      https://github.com/supabase/supabase/assets/8291514/49f6aa38-b28e-4ae0-a320-ba6e68ccf768
    - ~~You can now click on any item dropdown in the list without losing the active SQL snippet being shown on the right.~~
    - ~~also updated dropdowns to have better focus styling~~
      image
- ~~The animation for HoverCard can be updated when another PR gets merged in that has the animation styling in~~

### CODE REVIEW

**Comment by MildTomato** [comment]
This relies of PR https://github.com/supabase/supabase/pull/21486 to merge first

**Comment by joshenlim** [comment]
Project queries is missing an empty state?

**Comment by MildTomato** [comment]
back to drafts as I'm going to roll back the functional changes

File: apps/studio/components/layouts/SQLEditorLayout/QueryItemActions.tsx [hunk 1]
```diff
@@ -106,49 +100,57 @@ export const QueryItemActions = ({
...
+            name="Query actions"
+          >
             <Button
               type="text"
-              className="px-1 opacity-50 hover:opacity-100"
-              icon={<IconChevronDown size="tiny" strokeWidth={2} />}
+              className="px-1 text-lighter data-[state=open]:text-foreground"
+              icon={<MoreHorizontal size={14} strokeWidth={2} />}
+              onClick={(e) => {
+                console.log('clicked')
```
Comment (joshenlim) [inline apps/studio/components/layouts/SQLEditorLayout/QueryItemActions.tsx:116]: sneaky

File: apps/studio/components/layouts/SQLEditorLayout/QueryItem.tsx [hunk 2]
```diff
@@ -107,71 +112,58 @@ const QueryItem = ({
...
-                      ? 'group-hover:ml-5'
-                      : ''
-                )}
-              >
-                {name}
-              </Link>
-              {!hasQueriesSelected && (
-                <QueryItemActions
-                  tabInfo={tabInfo}
-                  activeId={activeId}
```
Comment (joshenlim) [inline apps/studio/components/layouts/SQLEditorLayout/QueryItem.tsx:155]: should remove the prop too

File: apps/studio/components/layouts/SQLEditorLayout/QueryItem.tsx [hunk 3]
```diff
@@ -19,6 +22,7 @@ import {
   TooltipContent_Shadcn_,
   TooltipTrigger_Shadcn_,
```
Comment (joshenlim) [inline apps/studio/components/layouts/SQLEditorLayout/QueryItem.tsx:23]: all these imports are no longer used right?

File: apps/studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx [hunk 4]
```diff
@@ -45,6 +50,11 @@ const SideBarContent = observer(() => {
   const project = useSelectedProject()
 
   const [searchText, setSearchText] = useState('')
+
+  // const [searchTextFavorite, setSearchTextFavoriteState] = useState('')
```
Comment (joshenlim) [inline apps/studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx:54]: remove?

**Review by joshenlim — APPROVED** [review]

**Pull request merged**